src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildTargetTFMSelector.cs

```diff
@@ -68,36 +69,53 @@ public TargetFrameworkMoniker SelectTFM(IProject project)
             var tfm = new TargetFrameworkMoniker(tfmName);
             foreach (var dep in project.ProjectReferences)
             {
-                try
+                foreach (var depTFM in dep.TargetFrameworks)
                 {
-                    if (_tfmComparer.IsCompatible(tfm, dep.TFM))
+                    try
                     {
-                        continue;
+                        if (_tfmComparer.IsCompatible(tfm, depTFM))
+                        {
+                            continue;
+                        }
+
+                        if (depTFM.IsNetCore || depTFM.IsNetStandard)
```

Comment:
This should only happen if none of the depTFMs are compatible with tfm.

Probably the logic here now should be:

```
foreach (var dep in project.ProjectReferences
{
    if (dep.TargetFrameworks.Any(IsCompatible)) continue;
    if (dep.TargetFrameworks.Any(IsNetCore || IsNetStandard)) tfm = the netcore/netstandard one
}
```

Reply:
Cleaned this up and added some tests. Switched to use `TargetFrameworkComparer.Compare` and it simplifies the code